soc/st/stm32/stm32h7/Kconfig.defconfig.stm32h745xx

```diff
@@ -0,0 +1,20 @@
+# ST STM32H745X MCU configuration options
+
+# Copyright (c) 2020 Alexander Kozhinov <AlexanderKozhinov@yandex.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32H745XX_M7 || SOC_STM32H745XX_M4
+
+config CPU_CORTEX_M7
+	default y if SOC_STM32H745XX_M7
+
+config CPU_CORTEX_M4
+	default y if SOC_STM32H745XX_M4
```

Comment:
would like @tejlmand to comment on this, we are doing this in `Kconfig` for every other board as a select

Reply:
This was indeed my initial version. I finally opted for this scheme as there files are there for SoC specific configuration

Reply:
@tejlmand This is blocking stm32u5 and stm32wba migration

Reply:
@nordicjm PTAL

Reply:
> @tejlmand This is blocking stm32u5 and stm32wba migration

I see code has changed to `select`, which would also be my preference.

The `CPU_CORTEX_Mx` settings are promptless and thus not suffering the stuck symbol syndrome.
Also it's tightly coupled to the SoC selection, so make sense to use select.

Reply:
Ok, we'll keep it that way then. Thanks